### ISSUE TITLE
Boot failed with amd64 artifact using uml backend

### ISSUE THREAD

**Issue by thanhtunng** [go-debos/debos#263]
Hí, 

Using uml backend to build seems OK, but i can not run amd64 artifact. 
```
Booting from Hard Disk...
error: no such device: 39ee5558-6459-45c6-8df0-1ba1591d2726.
error: unknown filesystem.
Entering rescue mode...
grub rescue>
```
although it matches uuid
```
/dev/loop11: PTUUID="da693184" PTTYPE="dos"
/dev/loop11p1: LABEL="ROOTFS" UUID="39ee5558-6459-45c6-8df0-1ba1591d2726" TYPE="ext4" PARTUUID="da693184-01"
```
I used this example https://github.com/sangorrin/debos-tests-fsuuid/blob/master/fsuuid.yaml.
It works with kvm backend.

Please let me know if you have any comments/ideas.

Thanks,
Tung

**Comment by obbardc** [comment]
So it just fails with the amd64 artefact built using UML ?

The fsuuid patches aren't yet merged inside the main branch so I am sure it would be going wrong if you are using the `godebos/debos` container :-)

**Comment by thanhtunng** [comment]
> So it just fails with the amd64 artefact built using UML ?

Yes, only amd64. Build for ARM architecture still OK with both kvm and uml.

> The fsuuid patches aren't yet merged inside the main branch so I am sure it would be going wrong if you are using the godebos/debos container :-)

Just an example, i tried several other ones and all of them don't work with only amd64.

Emphasized: issue with only amd64 built by uml =D other cases all works fine.

Thanks

**Comment by sangorrin** [comment]
@thanhtunng san did you try with the same example, but removing the fsuuid part?

**Comment by sangorrin** [comment]
Hi @obbardc, sorry we will review this internally and let you know if the problem is really in Debos or in the test code

**Comment by obbardc** [comment]
no problem, always happy to help ;-)

**Comment by sangorrin** [comment]
Hi @obbardc,

I have uploaded an example that reproduces the problem
- https://github.com/sangorrin/debos-tests-uml-amd64

We have noticed that the files under /boot/grub/grub.cfg have small differences (e.g. "insmod part_msdos" does not appear when using UML), but we still don't know the root of the problem.  We are guessing that grub-install or update-grub behaves differently when using UML. If we find something else, we will let you know.

Thanks,
Daniel

**Comment by sjoerdsimons** [comment]
I've had issues with grub various times (not just in UML); grub-install seem very senstive to the host environment. Unsure if there is much we can do about it not installing the msdos module in UML, would need to analyse why it behaves like that first :). Unsure if you got to that point already.

**Comment by sangorrin** [comment]
@sjoerdsimons thanks for your comment. It seems that this requires a deep dive and we can work around the issue by not installing grub legacy on our CI (i.e. using only an UEFI partition). For that reason, let me close the issue.
